clusterloader2/pkg/imagepreload/manifests/daemonset.yaml

```diff
@@ -15,16 +15,21 @@ spec:
       initContainers:
       {{range $idx, $image := .Images}}
       - name: preload-{{$idx}}
-        image: docker
-        command: ["docker", "pull", "{{$image}}"]
+        image: gcr.io/k8s-testimages/perf-tests-util/containerd:v0.0.1
+        command: ["sh", "-c", "ctr image pull {{$image}} 2>&1 | tee /var/log/cl2-image-preload-{{$idx}}.log"]
```

Comment:
I think I got it. We need to pull the image in the k8s.io namespace for the kubelet to update the node's image list. So, the command should be
```
ctr -n=k8s.io image pull ...
```

Could you change it?

Reply:
done. are containerd namespaces equivalent to k8s namespaces?

Reply:
Depending on what you mean by equivalent :)
The idea is the same - provide logical grouping of objects. But there is no mapping between k8s namespace and containerd namespaces. All images used in kubernetes are simply put into k8s.io containerd namespace.

Reply:
> All images used in kubernetes are simply put into k8s.io containerd namespace.

ah, saw that in the main repo containerd package. it's clear to me now 👍

Reply:
```
Found the following images on node: [gcr.io/k8s-testimages/probes:latest gcr.io/k8s-testimages/perf-tests-util/containerd@sha256:5ea41e1e29801ff8d07865c862f13a380a838f6e94b90f529aca7343e087e705 gcr.io/k8s-testimages/perf-tests-util/containerd:v0.0.1 k8s.gcr.io/kube-proxy-amd64:v1.21.0-alpha.1.63_083290d996cd07 k8s.gcr.io/prometheus-to-sd@sha256:14666989f40bb7c896c3e775a93c6873e2b791d65bc65579f58a078b7f9a764e k8s.gcr.io/prometheus-to-sd:v0.5.0 k8s.gcr.io/metadata-proxy@sha256:e914645f22e946bce5165737e1b244e0a296ad1f0f81a9531adc57af2780978a k8s.gcr.io/metadata-proxy:v0.1.12 k8s.gcr.io/pause@sha256:927d98197ec1141a368550822d18fa1c60bdae27b78b0c004f705f548c07814f k8s.gcr.io/pause:3.2]
```
🎉

Reply:
Nice! Let's revert then all the changes except daemonset.yaml and squash the commits into one and submit :)

Reply:
cool its done. when / where will we actually use the image preloader?

Reply:
We'll use it as soon as it's implemented to improve our scale tests. One place is kubemark which currently has no node-images which makes the size of the node object in kubemark completely not realistic (<<1KB). Another place are the regular scale tests where there are so node images but very few. In the end, the node object size there is ~1KB while in production clusters it's usually >10KB. It has an impact of the system performance and it's a gap in our e2e tests that we should close.